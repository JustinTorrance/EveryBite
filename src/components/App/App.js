import React, { Component } from 'react'
import IngredientsForm from '../../containers/IngredientsForm/IngredientsForm'
import CardContainer from '../../containers/CardContainer/CardContainer'
import Typography from '@material-ui/core/Typography'

export class App extends Component {
  render() {
    return (
      <div className="App">
    
          <div className='input-container'>
            <Typography variant="title" color="inherit">
              <IngredientsForm />
            </Typography>
          </div>
          <div className='recipe-container'>
            <Typography variant="title" color="inherit">
              <CardContainer />
            </Typography>
          </div>
      </div>
    );
  }
}

export default App
