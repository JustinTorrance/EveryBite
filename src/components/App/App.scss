* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100vh;
  overflow: scroll;
}

.title {
  font-family: 'Pacifico', cursive;
  margin-bottom: 20px;
  font-size: 3rem;
}