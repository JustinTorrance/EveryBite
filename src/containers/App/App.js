import React, { Component } from 'react'
import IngredientsForm from '../IngredientsForm/IngredientsForm'
import CardContainer from '../CardContainer/CardContainer'
import { Route, Switch } from 'react-router-dom'
import { connect } from 'react-redux'
import { Loading } from '../../components/Loading/Loading'
import PropTypes from 'prop-types'

export class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className='title'>Every<span>Bite</span></h1>
        { this.props.isLoading && <Loading /> }
        <Switch>
          <Route exact path='/' component={IngredientsForm} />
          <Route exact path='/recipes' component={CardContainer} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  isLoading: PropTypes.bool
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading.isRequired
})

export default connect(mapStateToProps)(App)
