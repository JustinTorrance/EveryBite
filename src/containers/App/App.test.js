import { App, mapStateToProps} from './App'

describe('App', () => {
  
  describe('mapStateToProps', () => {
    
    it('should return an object with isLoading and error', () => {
      const mockState = {
        isLoading: false,
        error: 'Something went wrong',
        recipes: [{name: 'curry'}]
      }
      const expected = {
        isLoading: false,
        error: 'Something went wrong'
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})