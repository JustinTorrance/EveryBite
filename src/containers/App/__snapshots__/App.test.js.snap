// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
"<div className=\\"App\\">
  <h1 className=\\"title\\">
    Every
    <span>
      Bite
    </span>
  </h1>
  <Switch>
    <Route exact={true} path=\\"/\\" component={[Function: C]} />
    <Route exact={true} path=\\"/recipes\\" component={{...}} />
    <Route component={[Function: BadPath]} />
  </Switch>
</div>"
`;
